src/render/View3D.cpp

```diff
@@ -498,6 +514,42 @@ void View3D::paintGL()
     glCheckError();
 }
 
+
+QImage View3D::renderScreenShot(int width, int height)
+{
+    if (m_badOpenGL)
+        return QImage();
+
+    double dPR = getDevicePixelRatio();
```

Comment:
As I understand things, this shouldn't be required, as `getDevicePixelRatio()` is related to the physical screen, not an offscreen buffer.

Instead, I'd suggest explicitly antialiasing, either by using `QOpenGLFramebufferObjectFormat`, or possibly more portably by just using a multiple of the actual width and height and downsampling on the CPU side (as you already have below with `scaleToWidth()`)